hunts/src/tagFilter.js

```diff
@@ -0,0 +1,24 @@
+function getDescendants(tableRow) {
+  const allChildren = [tableRow];
+  for (let i = 0; i < allChildren.length; i++) {
+    const current = allChildren[i];
+    Array.prototype.push.apply(allChildren, current.subRows);
+  }
+  return allChildren;
+}
+
+export function addOrRemoveTag(tagList, tag) {
+  let tagInList = tagList.filter(x => ['name', 'color', 'id'].every(key => x[key] === tag[key]))[0];
```

Comment:
Can we use `tagList.find(...)` instead of `tagList.filter(...)[0]` here?

Reply:
Yeah, didn't know find existed. (Also I'm a bit worried about this code, it uses the little-known fact that === on objects is "are these the same object" rather than "always false" or "do these have the same key-value pairs")

Reply:
Doesn't `===` just check for type and value equivalence (not necessarily the same object in memory)? If so, this should be okay, yeah?

Reply:
No, it checks for same object in memory, which is (at least part of) why we need to do this tagInList thing rather than just indexOf and includes.

Reply:
Yeah, it seems like `===` checks for object equivalence for non-primitive type objects, but for numbers and strings, it just does value comparison (lemme know if I'm misinterpreting): http://www.ecma-international.org/ecma-262/5.1/#sec-11.9.6

Are the `name`, `color`, and `id` fields not just strings/numbers?

Reply:
They are just strings and numbers, and so equality is by value rather than memory, I think I was being unclear here. I'm talking about the use of !== in the filter on the next line:
`  return (tagInList !== undefined) ? tagList.filter((i) => i !== tagInList) : tagList.concat(tag);`,
which only works because === does compare identity, rather than just always being false.

If `===` were always value identity I think I could do something like
`  return tagList.includes(tag) ? tagList.filter((i) => i !== tag) : tagList.concat(tag);`
without this whole tagInList business, which would've been simpler.